wdl/finemap.wdl

```diff
@@ -30,7 +31,58 @@ task preprocess {
     Float? minimum_pval
     String? set_variant_id_map_chr
 
-    command {
+    command <<<
+
+        catcmd="cat"
+        if [[ ${phenofile} == *.gz ]] || [[ ${phenofile} == *.bgz ]]
+        then
+            catcmd="zcat"
+        fi
+
+        echo "Reading phenotype file with $catcmd"
+        $catcmd ${phenofile} | awk -v ph=${pheno} '
+        BEGIN {
+            FS = "\t"
+        }
+        NR == 1 {
+            for(i = 1; i <= NF; i++) {
+                h[$i] = i
+            }
+            exists=ph in h
+            if (!exists) {
+                print "Phenotype:"ph" not found in the given phenotype file." > "/dev/stderr"
+                err = 1
+                exit 1
+            }
+        }
+        NR > 1 && $h[ph] != "NA" {
+            vals[$h[ph]] += 1
+            print $h["FINNGENID"] > ph".incl"
+            if ($h[ph] != 0 && $h[ph] != 1) {
+                print "Phenotype:"ph" seems a quantitative trait. Setting var_y = 1 and prior_std = 0.05." > "/dev/stderr"
+                print 1.0 > "var_y.txt"
+                print 0.05 > "prior_std.txt"
+                err = 1
+                exit 0
+            }
+        }
+        END {
+            if (!err) {
+                phi = vals["1"] / (vals["1"]+vals["0"])
+                var_y = phi * (1-phi)
+                std = 0.05 * sqrt(phi*(1-phi))
+                printf var_y > "var_y.txt"
```

Comment:
printf here and not others does not affect anything except missing newline but looks suspicious to autist like me :)

Reply:
great catch! addressed